File: backend/PermissionSystem.Domain/Entities/SsoUserBinding.cs
using PermissionSystem.Domain.Common;

namespace PermissionSystem.Domain.Entities;

public sealed class SsoUserBinding : BaseEntity
{
    public Guid ProviderId { get; set; }

    public string ProviderCode { get; set; } = string.Empty;

    public string ExternalUserId { get; set; } = string.Empty;

    public string? ExternalUserName { get; set; }

    public string? ExternalEmail { get; set; }

    public string? ExternalPhone { get; set; }

    public Guid LocalUserId { get; set; }

    public DateTimeOffset? LastLoginAt { get; set; }

    public string? ClaimsJson { get; set; }

    public SsoProvider? Provider { get; set; }

    public User? LocalUser { get; set; }
}

File: frontend/permission-admin/src/api/ssoUserBinding.ts
import { request } from '../utils/request'
import type { ApiResult, PagedResult, PageQuery } from './types'

export interface SsoUserBindingQuery extends PageQuery {
  providerId?: string
}

export interface SsoUserBindingItem {
  id: string
  tenantId: string
  providerId: string
  providerCode: string
  providerName?: string
  externalUserId: string
  externalUserName?: string
  externalEmail?: string
  externalPhone?: string
  localUserId: string
  localUserName?: string
  localDisplayName?: string
  lastLoginAt?: string
  createdAt: string
}

export interface SsoUserBindingDetail extends SsoUserBindingItem {
  claimsJson?: string
}

export function getSsoUserBindings(params: SsoUserBindingQuery) {
  return request
    .get<ApiResult<PagedResult<SsoUserBindingItem>>>('/api/sso/user-bindings', { params })
    .then((res) => res.data.data)
}

export function getSsoUserBinding(id: string) {
  return request
    .get<ApiResult<SsoUserBindingDetail>>(`/api/sso/user-bindings/${id}`)
    .then((res) => res.data.data)
}

export function unbindSsoUser(id: string) {
  return request.post<ApiResult<void>>(`/api/sso/user-bindings/${id}/unbind`)
}

export function deleteSsoUserBinding(id: string) {
  return request.delete<ApiResult<void>>(`/api/sso/user-bindings/${id}`)
}
